### DISCUSSION TITLE
Questions about Copilot + Open Source Software Hierarchy

### DISCUSSION THREAD

**Discussion by liaochris** [pyinstaller/pyinstaller#8371] [PyInstaller] [answered]
Redirected from: [https://github.com/pyinstaller/pyinstaller/issues/8370#issuecomment-2016883923](https://github.com/pyinstaller/pyinstaller/issues/8370#issuecomment-2016883923)

Hi! My name is Chris and I'm writing a thesis on Open Source Software. I'm trying to collect/validate my data and I have two questions for the maintainers of this project.

1. Did this project receive free github copilot access on June 2022? Alternatively, you can also see if the project currently has free copilot access by going to this link (if you have write access or above): [https://github.com/github-copilot/free_signup](https://github.com/github-copilot/free_signup)
2. My thesis is especially focused on understanding hierarchical structures. Would it be possible to share a list of individuals in this project with triage/write/maintain/admin access in this project, if it's publicly available?

Happy to chat further if you have questions, and thank you for your time!

**Comment by bwoodsend** [comment, marked as answer]
> Did this project receive free github copilot access on June 2022? Alternatively, you can also see if the project currently has free copilot access by going to this link (if you have write access or above): https://github.com/github-copilot/free_signup

No, and I doubt we'd have bothered trying it if we had. Two out of three of us are so minimalist that we can't even stomach PyCharm – preferring to use geany. We also get no end of people reporting completely nonsense bugs because ChatGPT told them to do something very stupid and it unsurprisingly didn't work so we're all pretty sceptical about AI writing code. 

> My thesis is especially focused on understanding hierarchical structures. Would it be possible to share a list of individuals in this project with triage/write/maintain/admin access in this project, if it's publicly available?

Myself, rokm and Legorooj are admins. We do have a ladder of triage/write roles although they're long unused as it's been years since we gained a new maintainer.

**Reply by liaochris** [reply]
Great, thanks for your response! Is it accurate to say that most maintenance work is done by the admins (Myself, rokm and Legorooj)? Or are the people in the write/triage roles still pretty involved/active to this day?

**Reply by bwoodsend** [reply]
There are no other non-admin people in the write/triage roles – that's what I meant by their being unused.

Here are the commit counts for the last year:
```
   382	Rok Mandeljc         # a.k.a rokm 
    47	Brénainn Woodsend    # me
     7	Dan Yeaw
     2	Hugo van Kemenade
     2	Michael Shigorin
     1	Amir Rossert
     1	Andrey Marakulin
     1	Benedikt Würkner
     1	Blank
     1	Brandon
     1	Christian Clauss
     1	David Baumgold
     1	Fabian Dröge
     1	Geo5
     1	Ievgen Popovych
     1	James Gerity
     1	Joshua Bronson
     1	Levin Ma
     1	Lorenzo Villani
     1	Olliver Aikenhead
     1	RoboSchmied
     1	Sebastian Thomschke
     1	V. Armando Solé
     1	axoroll7
     1	byehack
     1	caption
     1	cat
     1	coolcatco888
     1	eduardomotta-emottasistemas
     1	gentlegiantJGC
     1	mbushkov
     1	xuanzhi33
```

**Reply by liaochris** [reply]
Got it, thanks for clarifying!
